tests/spec_decode/e2e/conftest.py

```diff
@@ -1,10 +1,132 @@
-from typing import List, Tuple
+import asyncio
+from typing import List, Optional, Tuple, Union
 
 import pytest
 
 from tests.conftest import cleanup
 from vllm import LLM
+from vllm.engine.arg_utils import AsyncEngineArgs, EngineArgs
+from vllm.engine.async_llm_engine import AsyncLLMEngine
+from vllm.engine.llm_engine import LLMEngine
+from vllm.lora.request import LoRARequest
 from vllm.model_executor.utils import set_random_seed
+from vllm.outputs import RequestOutput
+from vllm.sampling_params import SamplingParams
+from vllm.sequence import MultiModalData
+from vllm.usage.usage_lib import UsageContext
+from vllm.utils import Counter, random_uuid
+
+"""AsyncLLM
+
+Note: Crrurent LLM class in vllm don't support async mode, for test purpose,
+we implement async one in here. Maybe we could move to vllm/entrypoints/llm.py
+in future.
+
+Below AsyncLLM is directly borrow from vllm/entrypoints/llm.py with changes
+to make to work in async mode.
+"""
+
+
+class AsyncLLM:
```

Comment:
Consider supporting gpu cleanup using `cleanup` method in conftest.py? 

I think you can make it a conftest and delete the instance & call cleanup() in the destructor simliar to https://github.com/vllm-project/vllm/blob/468d761b32e3b3c5d64eeaa797e54ab809b7e50c/tests/conftest.py#L275